crates/time/src/lib.rs

```diff
@@ -0,0 +1,230 @@
+type Slot = u64;
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct Bound {
+    pub time: u64, // Milliseconds
+    pub slot: Slot,
+    pub epoch: u64,
+}
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct EraParams {
+    epoch_size: u64,
+    slot_length: u64, // Milliseconds
+}
+
+impl EraParams {
+    pub fn new(epoch_size: u64, slot_length: u64) -> Option<Self> {
+        if epoch_size == 0 {
+            return None;
+        }
+        if slot_length == 0 {
+            return None;
+        }
+        Some(EraParams {
+            epoch_size,
+            slot_length,
+        })
+    }
+}
+
+// The start is inclusive and the end is exclusive. In a valid EraHistory, the
+// end of each era will equal the start of the next one.
```

Comment:
Is this another constraint it makes sense to validate / enforce in an `EraHistory` constructor @KtorZ?

i.e. just pass in a `Vec<(Bound, Params)>`, and then use the `start` of the next one as the `end` of the previous one, with the last one taking a `None`?